Kompose fails on version 3 with deploy-resources-limits (panic: runtime error: invalid memory address or nil pointer dereference)

Using kompose version 1.10.0 (8bb0907) on Ubuntu 16.04 and converting using 
```
kompose convert -f b.yaml
```

-----

b.yaml:

```
services:
  haproxydocker:
    command: haproxy -f /etc/haproxy/haproxy_dsp.cfg
    environment:
      BACKEND_1_IP: 192.168.1.105
    image: haproxy
    network_mode: host
    restart: always
    deploy:
      resources:
        limits:
          memory: 256m
version: "3"
```
---------
I get following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1622d57]

goroutine 1 [running]:
github.com/kubernetes/kompose/pkg/loader/compose.dockerComposeToKomposeMapping(0xc420010600, 0x1c, 0xc420133c20, 0x1, 0x1, 0xc4201a5e60)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/pkg/loader/compose/v3.go:267 +0xfe7
github.com/kubernetes/kompose/pkg/loader/compose.parseV3(0xc42052db70, 0x1, 0x1, 0x1, 0x1, 0x0, 0x30, 0x1bec580)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/pkg/loader/compose/v3.go:113 +0x22b
github.com/kubernetes/kompose/pkg/loader/compose.(*Compose).LoadFile(0x2b1f0d8, 0xc42052db70, 0x1, 0x1, 0xc4201a5830, 0x0, 0x1a, 0x7f6a9adfcd50, 0x454d10)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/pkg/loader/compose/compose.go:186 +0x4d2
github.com/kubernetes/kompose/pkg/app.Convert(0x100000100, 0x0, 0x0, 0x0, 0x0, 0x1cac29c, 0x4, 0x0, 0x1cc0501, 0x15, ...)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/pkg/app/app.go:232 +0x145
github.com/kubernetes/kompose/cmd.glob..func3(0x2aec8e0, 0xc42045f900, 0x0, 0x2)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/cmd/convert.go:95 +0x4d
github.com/kubernetes/kompose/vendor/github.com/spf13/cobra.(*Command).execute(0x2aec8e0, 0xc42045f8c0, 0x2, 0x2, 0x2aec8e0, 0xc42045f8c0)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/vendor/github.com/spf13/cobra/command.go:704 +0x2c6
github.com/kubernetes/kompose/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2aecd20, 0x2aed160, 0x18a1b60, 0x2aece60)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/vendor/github.com/spf13/cobra/command.go:785 +0x30e
github.com/kubernetes/kompose/vendor/github.com/spf13/cobra.(*Command).Execute(0x2aecd20, 0x0, 0xc420275f48)
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/vendor/github.com/spf13/cobra/command.go:738 +0x2b
github.com/kubernetes/kompose/cmd.Execute()
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/cmd/root.go:92 +0x31
main.main()
	/home/wikus/seafile/files/dev/go/src/github.com/kubernetes/kompose/main.go:22 +0x20
```
-----

If I comment out deploy section:
```
    # deploy:
      # resources:
        # limits:
          # memory: 256m
```
it works OK and creates deployment and service. yaml

To be sure that this is docker-compose.yaml (b.yaml) is valid synthax I also checked with
docker-compose -f b.yaml config
and it works OK (used docker-compose version 1.19.0, build 9e633ef)

--- a/pkg/loader/compose/v3.go
+++ b/pkg/loader/compose/v3.go
@@ -262,27 +262,34 @@ func dockerComposeToKomposeMapping(composeObject *types.Config) (kobject.Kompose
 
 			// memory:
 			// TODO: Refactor yaml.MemStringorInt in kobject.go to int64
-			// Since Deploy.Resources.Limits does not initialize, we must check type Resources before continuing
-			serviceConfig.MemLimit = libcomposeyaml.MemStringorInt(composeServiceConfig.Deploy.Resources.Limits.MemoryBytes)
-			serviceConfig.MemReservation = libcomposeyaml.MemStringorInt(composeServiceConfig.Deploy.Resources.Reservations.MemoryBytes)
-
 			// cpu:
 			// convert to k8s format, for example: 0.5 = 500m
 			// See: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/
 			// "The expression 0.1 is equivalent to the expression 100m, which can be read as “one hundred millicpu”."
 
-			cpuLimit, err := strconv.ParseFloat(composeServiceConfig.Deploy.Resources.Limits.NanoCPUs, 64)
-			if err != nil {
-				return kobject.KomposeObject{}, errors.Wrap(err, "Unable to convert cpu limits resources value")
-			}
-			serviceConfig.CPULimit = int64(cpuLimit * 1000)
+			// Since Deploy.Resources.Limits does not initialize, we must check type Resources before continuing
+			if composeServiceConfig.Deploy.Resources.Limits != nil {
+				serviceConfig.MemLimit = libcomposeyaml.MemStringorInt(composeServiceConfig.Deploy.Resources.Limits.MemoryBytes)
 
-			cpuReservation, err := strconv.ParseFloat(composeServiceConfig.Deploy.Resources.Reservations.NanoCPUs, 64)
-			if err != nil {
-				return kobject.KomposeObject{}, errors.Wrap(err, "Unable to convert cpu limits reservation value")
+				if composeServiceConfig.Deploy.Resources.Limits.NanoCPUs != "" {
+					cpuLimit, err := strconv.ParseFloat(composeServiceConfig.Deploy.Resources.Limits.NanoCPUs, 64)
+					if err != nil {
+						return kobject.KomposeObject{}, errors.Wrap(err, "Unable to convert cpu limits resources value")
+					}
+					serviceConfig.CPULimit = int64(cpuLimit * 1000)
+				}
 			}
-			serviceConfig.CPUReservation = int64(cpuReservation * 1000)
+			if composeServiceConfig.Deploy.Resources.Reservations != nil {
+				serviceConfig.MemReservation = libcomposeyaml.MemStringorInt(composeServiceConfig.Deploy.Resources.Reservations.MemoryBytes)
 
+				if composeServiceConfig.Deploy.Resources.Reservations.NanoCPUs != "" {
+					cpuReservation, err := strconv.ParseFloat(composeServiceConfig.Deploy.Resources.Reservations.NanoCPUs, 64)
+					if err != nil {
+						return kobject.KomposeObject{}, errors.Wrap(err, "Unable to convert cpu limits reservation value")
+					}
+					serviceConfig.CPUReservation = int64(cpuReservation * 1000)
+				}
+			}
 		}
 
 		// restart-policy: deploy.restart_policy.condition will rewrite restart option

--- a/script/test/cmd/tests.sh
+++ b/script/test/cmd/tests.sh
@@ -498,6 +498,10 @@ cmd="kompose convert --stdout -j -f $KOMPOSE_ROOT/script/test/fixtures/v3/docker
 sed -e "s;%VERSION%;$version;g" -e "s;%CMD%;$cmd;g"  "$KOMPOSE_ROOT/script/test/fixtures/v3/output-memcpu-k8s.json" > /tmp/output-k8s.json
 convert::expect_success "kompose convert --stdout -j -f $KOMPOSE_ROOT/script/test/fixtures/v3/docker-compose-memcpu.yaml" "/tmp/output-k8s.json"
 
+cmd="kompose convert --stdout -j -f $KOMPOSE_ROOT/script/test/fixtures/v3/docker-compose-memcpu-partial.yaml"
+sed -e "s;%VERSION%;$version;g" -e "s;%CMD%;$cmd;g"  "$KOMPOSE_ROOT/script/test/fixtures/v3/output-memcpu-partial-k8s.json" > /tmp/output-k8s.json
+convert::expect_success "$cmd" "/tmp/output-k8s.json"
+
 # Test volumes are passed correctly
 cmd="kompose convert --stdout -j -f $KOMPOSE_ROOT/script/test/fixtures/v3/docker-compose-volumes.yaml"
 sed -e "s;%VERSION%;$version;g" -e "s;%CMD%;$cmd;g"  $KOMPOSE_ROOT/script/test/fixtures/v3/output-volumes-k8s-template.json > /tmp/output-k8s.json

--- a/script/test/fixtures/v3/docker-compose-memcpu-partial.yaml
+++ b/script/test/fixtures/v3/docker-compose-memcpu-partial.yaml
@@ -1,0 +1,11 @@
+version: "3"
+
+services:
+  foo:
+    deploy:
+      resources:
+        limits:
+          memory: 50M
+        reservations:
+          cpus: '0.001'
+    image: redis

--- a/script/test/fixtures/v3/output-memcpu-partial-k8s.json
+++ b/script/test/fixtures/v3/output-memcpu-partial-k8s.json
@@ -1,0 +1,83 @@
+{
+  "kind": "List",
+  "apiVersion": "v1",
+  "metadata": {},
+  "items": [
+    {
+      "kind": "Service",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "foo",
+        "creationTimestamp": null,
+        "labels": {
+          "io.kompose.service": "foo"
+        },
+        "annotations": {
+          "kompose.cmd": "%CMD%",
+          "kompose.version": "%VERSION%"
+        }
+      },
+      "spec": {
+        "ports": [
+          {
+            "name": "headless",
+            "port": 55555,
+            "targetPort": 0
+          }
+        ],
+        "selector": {
+          "io.kompose.service": "foo"
+        },
+        "clusterIP": "None"
+      },
+      "status": {
+        "loadBalancer": {}
+      }
+    },
+    {
+      "kind": "Deployment",
+      "apiVersion": "extensions/v1beta1",
+      "metadata": {
+        "name": "foo",
+        "creationTimestamp": null,
+        "labels": {
+          "io.kompose.service": "foo"
+        },
+        "annotations": {
+          "kompose.cmd": "%CMD%",
+          "kompose.version": "%VERSION%"
+        }
+      },
+      "spec": {
+        "replicas": 1,
+        "template": {
+          "metadata": {
+            "creationTimestamp": null,
+            "labels": {
+              "io.kompose.service": "foo"
+            }
+          },
+          "spec": {
+            "containers": [
+              {
+                "name": "foo",
+                "image": "redis",
+                "resources": {
+                  "limits": {
+                    "memory": "52428800"
+                  },
+                  "requests": {
+                    "cpu": "1m"
+                  }
+                }
+              }
+            ],
+            "restartPolicy": "Always"
+          }
+        },
+        "strategy": {}
+      },
+      "status": {}
+    }
+  ]
+}

